### PR TITLE
Add legacy vs experimental ORTTrainer accuracy comparison

### PR DESCRIPTION
**Pull request by rayankrish**
- Adds a relevant helper function for comparing weights.

- Adds test for legacy vs experimental weights to ensure train step consistency across the new API.

### CODE REVIEW

File: orttraining/orttraining/python/training/debug.py [hunk 1]
```diff
@@ -21,11 +22,30 @@ def compare_onnx_weights(model_a, model_b, verbose=False, rtol=1e-4):
...
+
+    Compares the weights of two different ONNX models and throws an error when they diverge
+    Args:
+        model_a, model_b (ORTTrainer): Two instances of ORTTrainer with the same model structure
+        verbose (bool, default is False): Indicates if the max absolute difference for each layer should be
+    calculated and printed for debug information.
+        rtol (float, default is 1e-4): Tolerance for divergence.
+    """
+    assert isinstance(model_a, orttrainer.ORTTrainer) and isinstance(model_b, Legacy_ORTTrainer)
+    state_dict_a, state_dict_b = model_a._training_session.get_state(), model_b.session.get_state()
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/debug.py:39]:
```suggestion
    # TODO: update to the new API once checkpoint is implemented
    state_dict_a, state_dict_b = model_a._training_session.get_state(), model_b.session.get_state()
```

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py [hunk 2]
```diff
@@ -473,7 +473,7 @@ def generate_pytorch_transformer_model_sample(optim_config, options={}, step_fn=
     utils = _utils.import_module_from_file(utils_path, utils_name)
 
     # Modeling
-    model = pt_model.TransformerModel(28785, 200, 2, 200, 2, 0.2)
+    model = pt_model.TransformerModel(28785, 200, 2, 200, 2, 0.2).to(device)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py:476]:
```suggestion
    model = pt_model.TransformerModel(28785, 200, 2, 200, 2, 0.2)
```

File: orttraining/orttraining/python/training/debug.py [hunk 3]
```diff
@@ -21,11 +22,30 @@ def compare_onnx_weights(model_a, model_b, verbose=False, rtol=1e-4):
     assert isinstance(model_a, orttrainer.ORTTrainer) and isinstance(model_b, orttrainer.ORTTrainer)
     state_dict_a, state_dict_b = model_a._training_session.get_state(), model_b._training_session.get_state()
     assert len(state_dict_a.items()) == len(state_dict_b.items())
+    _compare_state_dict_weights(state_dict_a, state_dict_b, verbose, rtol)
+
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/debug.py:26]:
```suggestion


```

File: samples/python/pytorch_transformer/ort_utils.py [hunk 4]
```diff
@@ -16,3 +17,12 @@ def transformer_model_description():
                   'outputs': [('loss', [], True),
                               ('predictions', [bptt, batch_size, ntokens])]}
     return model_desc
+
```
Comment (thiagocrepaldi) [inline samples/python/pytorch_transformer/ort_utils.py:20]:
```suggestion


```

File: samples/python/pytorch_transformer/ort_utils.py [hunk 5]
```diff
@@ -16,3 +17,12 @@ def transformer_model_description():
...
+def legacy_transformer_model_description():
+    input_desc = IODescription('input1', [35, 20], torch.float32)
+    label_desc = IODescription('label', [35, 20, 28785], torch.int64)
+    loss_desc = IODescription('loss', [], torch.float32)
+    lr = 0.001
+    #return ModelDescription([input_desc, label_desc], [loss_desc]), IODescription('Learning_Rate', [lr,], torch.float32)
+    prediction_desc = IODescription('prediction', [35, 20, 28785], torch.float32)
+    return ModelDescription([input_desc, label_desc], [loss_desc, prediction_desc]), IODescription('Learning_Rate', [lr,], torch.float32)
```
Comment (thiagocrepaldi) [inline samples/python/pytorch_transformer/ort_utils.py:28]:
```suggestion
    return ModelDescription([input_desc, label_desc], [loss_desc, prediction_desc]), IODescription('Learning_Rate', [lr,], torch.float32)

```
